### PR TITLE
[App Service]: `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError And `az webapp deleted restore` : Fix the issue caused by AttributeError

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
**Description**

#19210 , #19236 

`19210`: az webapp deployment source config-local-git command will fail if slot option is specified.
`19236`: az webapp deleted restore - 'WebAppsOperations' object has no attribute 'restore_from_deleted_app'
**Testing Guide**

`az webapp deleted restore --deleted-id "deleted-id" --name webappname --resource-group RGName -s slotname`
`az webapp deployment source config-local-git --name webappname --resource-group RGName -s slotname `
**History Notes**

[App Service]: `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified
[App Service]: `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app'

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 1]
```diff
@@ -1903,35 +1903,13 @@ def test_one_deploy(self, resource_group):
...
-        ).assert_with_checks([
-            JMESPathCheck('siteName', webapp_name)
-        ])
-         #create an empty slot
-        self.cmd('webapp deployment slot create -g {} -n {} --slot {} --debug'.format(resource_group, webapp_name, slot))
-        self.cmd("webapp config hostname add -g {} --webapp-name {} --slot {} --hostname {} ".format(
-            resource_group, webapp_name, slot, slothostname)
-        ).assert_with_checks([
-            JMESPathCheck('name', "{}/{}/{}".format(webapp_name, slot, slothostname))
-        ])
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:1934]: May I ask why did you delete these test commands?

Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:1934]: This test should suppose to remove in [this](https://github.com/Azure/azure-cli/pull/19348) pull request. But some how I got messed up  and missed to do. So just added here.

**Review by panchagnula — CHANGES_REQUESTED** [review]
Approved

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:1934]: OK

**Review by zhoxing-ms — APPROVED** [review]

**Review by panchagnula — APPROVED** [review]
Approved
